sequences.lisp

```diff
@@ -1935,3 +1935,184 @@ Repetitions that are not adjacent are left alone.
               (setf val (key-fn item) init t)
               (unless (test val (key-fn item))
                 (return-from same nil))))))))
+
+(declaim (inline copy-enough-list nsplice-list splice-list))
+
+(defun copy-enough-list (list n)
```

Comment:
Since we already have a `firstn` function, I think `(copy-firstn (n list)` would be a more consistent name/signature.